src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/ProcessLinkerXmlBase.cs

```diff
@@ -337,169 +278,61 @@ void ProcessSelectedMethods(XPathNavigator nav, TypeDefinition type, object? cus
             }
         }
 
-        protected virtual void ProcessMethod(TypeDefinition type, XPathNavigator nav, object? customData)
+        protected virtual void ProcessMethod(TypeDesc type, XPathNavigator nav, object? customData)
         {
             string signature = GetSignature(nav);
             if (!String.IsNullOrEmpty(signature))
             {
-                MethodDefinition? method = GetMethod(type, signature);
-                if (method == null)
-                {
-                    LogWarning(nav, DiagnosticId.XmlCouldNotFindMethodOnType, signature, type.GetDisplayName());
-                    return;
-                }
-
-                ProcessMethod(type, method, nav, customData);
+                foreach (MethodDesc meth in type.GetMethods())
```

Comment:
Why the change here from `MethodDefinition? method = GetMethod(type, signature);` to a `foreach`?